eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigReplacerProcessor.java

```diff
@@ -16,12 +16,20 @@ public ContigReplacerProcessor(ContigMapping contigMapping) {
 
     @Override
     public SubSnpNoHgvs process(SubSnpNoHgvs subSnpNoHgvs) throws Exception {
-        ContigSynonyms contigSynonyms = contigMapping.getContigSynonyms(subSnpNoHgvs.getContigName());
-        if (contigSynonyms != null) {
-            subSnpNoHgvs.setContigName(contigSynonyms.getSequenceName());
-            return subSnpNoHgvs;
+        ContigSynonyms contigSynonyms;
+        if (subSnpNoHgvs.getChromosome() != null) {
+            contigSynonyms = contigMapping.getContigSynonyms(subSnpNoHgvs.getChromosome());
+        } else {
+            contigSynonyms = contigMapping.getContigSynonyms(subSnpNoHgvs.getContigName());
         }
-        throw new IllegalArgumentException(
-                "Contig '" + subSnpNoHgvs.getContigName() + "' not found in the ASSEMBLY REPORT");
+
+        if (contigSynonyms == null) {
+            throw new IllegalArgumentException(
+                    "Contig '" + subSnpNoHgvs.getContigName() + "' not found in the assembly report");
+        }
+
+        subSnpNoHgvs.setChromosome(contigSynonyms.getSequenceName());
+        subSnpNoHgvs.setContigName(contigSynonyms.getSequenceName());
```

Comment:
Chromosome coordinates are used only if the chromosome name is not null. So in order to avoid changing behavior only those that are non-null should be overridden.